### PR TITLE
keytool: Remove default key scheme as ed25519

### PR DESCRIPTION
**Pull request by joyqvq**
now `sui keytool generate` and `sui client new-address` must specify key scheme type as argument {secp256k1 || ed25519}

### CODE REVIEW

File: crates/sui/src/sui_commands.rs
Comment (patrickkuo) [inline crates/sui/src/sui_commands.rs:411]: I know this is not from this PR but can we translate the flag to the actual key/curve name?(maybe next PR) Showing only the flag is not very informative to the user.

File: crates/sui-sdk/src/crypto.rs
Comment (patrickkuo) [inline crates/sui-sdk/src/crypto.rs:159]: can we use enums? We have `SignatureScheme` in `sui_type::crypto`

**Review by Clay-Mysten — COMMENTED** [review]
Thanks, Joy. The Markdown changes LGTM. This change seems significant enough to announce in the Learn > See what's new section:
https://docs.sui.io/devnet/learn#see-whats-new

Mind if I do that? Also, I am mentioning I don't see any references to `keytool generate` in our docs and wonder if there should be one? When is that done?

Thanks!

**Comment by Clay-Mysten** [comment]
> Thanks, Joy. The Markdown changes LGTM. This change seems significant enough to announce in the Learn > See what's new section: https://docs.sui.io/devnet/learn#see-whats-new
> 
> Mind if I do that? Also, I am mentioning I don't see any references to `keytool generate` in our docs and wonder if there should be one? When is that done?
> 
> Thanks!

I can tack onto the Crypto example PR:
https://github.com/MystenLabs/sui/pull/4357

Or create anew. Please let me know. Thanks!

**Comment by joyqvq** [comment]
> > Thanks, Joy. The Markdown changes LGTM. This change seems significant enough to announce in the Learn > See what's new section: https://docs.sui.io/devnet/learn#see-whats-new
> > Mind if I do that? Also, I am mentioning I don't see any references to `keytool generate` in our docs and wonder if there should be one? When is that done?
> > Thanks!
> 
> I can tack onto the Crypto example PR: #4357
> 
> Or create anew. Please let me know. Thanks!

keytool command is mostly used as utility to view/import/list sui.keystore or interact with the key with sign. the cli documentation here look comprehensive to me, you can add for what's new or add a new section for these commands in doc. 
```
Sui keystore tool

USAGE:
    sui keytool [OPTIONS] <SUBCOMMAND>

OPTIONS:
    -h, --help                             Print help information
        --keystore-path <KEYSTORE_PATH>

SUBCOMMANDS:
    generate        Generate a new keypair with keypair scheme flag {ed25519 | secp256k1}.
                        Output file to current dir (to generate keypair to sui.keystore, use `sui
                        client new-address`)
    help            Print this message or the help of the given subcommand(s)
    import          Import mnemonic phrase and generate keypair based on key scheme flag
                        {ed25519 | secp256k1}
    list            List all keys by its address, public key, key scheme in the keystore
    load-keypair    This is a temporary helper function to ensure that testnet genesis does not
                        break while we transition towards BLS signatures
    show
    sign            Create signature using the sui keystore and provided data
    unpack          Extract components of a base64-encoded keypair to reveal the Sui address,
                        public key, and key scheme flag
```

File: crates/sui/src/sui_commands.rs
Comment (kchalkias) [inline crates/sui/src/sui_commands.rs:408]: This works for now, but in the future I'd prefer we specifically request for 0 and 1 (instead of y/N). This is because 0, 1 can extend to support multiple key types (and y/N) is just a temp solutions (it happens to work just because we have two schemes only - but we'll soon might add seck256r1 (NIST curve))

File: crates/sui/src/sui_commands.rs
Comment (kchalkias) [inline crates/sui/src/sui_commands.rs:403]: This comment is not aligned with what 0 and 1 represent below: i.e. `0 for ed25519` but then
`"0" => SignatureScheme::Secp256k1`

**Comment by Clay-Mysten** [comment]
> > > Thanks, Joy. The Markdown changes LGTM. This change seems significant enough to announce in the Learn > See what's new section: https://docs.sui.io/devnet/learn#see-whats-new
> > > Mind if I do that? Also, I am mentioning I don't see any references to `keytool generate` in our docs and wonder if there should be one? When is that done?
> > > Thanks!
> > 
> > 
> > I can tack onto the Crypto example PR: #4357
> > Or create anew. Please let me know. Thanks!
> 
> keytool command is mostly used as utility to view/import/list sui.keystore or interact with the key with sign. the cli documentation here look comprehensive to me, you can add for what's new or add a new section for these commands in doc.
> 
> ```
> Sui keystore tool
> 
> USAGE:
>     sui keytool [OPTIONS] <SUBCOMMAND>
> 
> OPTIONS:
>     -h, --help                             Print help information
>         --keystore-path <KEYSTORE_PATH>
> 
> SUBCOMMANDS:
>     generate        Generate a new keypair with keypair scheme flag {ed25519 | secp256k1}.
>                         Output file to current dir (to generate keypair to sui.keystore, use `sui
>                         client new-address`)
>     help            Print this message or the help of the given subcommand(s)
>     import          Import mnemonic phrase and generate keypair based on key scheme flag
>                         {ed25519 | secp256k1}
>     list            List all keys by its address, public key, key scheme in the keystore
>     load-keypair    This is a temporary helper function to ensure that testnet genesis does not
>                         break while we transition towards BLS signatures
>     show
>     sign            Create signature using the sui keystore and provided data
>     unpack          Extract components of a base64-encoded keypair to reveal the Sui address,
>                         public key, and key scheme flag
> ```

Thanks, @joyqvq . This is great news. Here is a PR to announce this change:
https://github.com/MystenLabs/sui/pull/4389

I will not submit until you merge in this change. Please do take a look. Asking @kchalkias to too.

File: crates/sui-types/src/crypto.rs
Comment (kchalkias) [inline crates/sui-types/src/crypto.rs:459]: <3

File: crates/sui-types/src/crypto.rs
Comment (kchalkias) [inline crates/sui-types/src/crypto.rs:1342]: I'd suggest that you match against our flags https://github.com/MystenLabs/sui/blob/d352fc2a25320f8082a2960525f621aea4a6b5e1/crates/sui-types/src/crypto.rs#L1333 vs. hardcoding "0" and "1" here, so if we ever change the flags there, the keytool will automatically work with the new assignment. 
*You will just need byte to string conversion which is easy.

**Review by kchalkias — APPROVED** [review]

File: crates/sui-types/src/crypto.rs
Comment (joyqvq) [reply crates/sui-types/src/crypto.rs]: good point (i went back and forth on this bc it involves validating the string to strictly one byte), revised!

**Pull request merged**
